### PR TITLE
Add some basic docs

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: docs/concept-exercises.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Concept Exercises replace V2's Core Exercises. They are exercises designed to teach specific programming concepts.
+
+## What do we mean by programming concepts?
+
+By concepts we mean things that a programmer would need to understand to be fluent in a language. We care specifically about how languages are **different**. What do I need to learn differently about numbers in Haskell to numbers in Ruby to be able to work with numbers in those languages. Two questions that we have felt useful to ask to establish this are:
+
+- If someone learnt Ruby, and someone learnt Haskell, what are the things that the two people learnt that are different?
+- If a Ruby programmer learnt Haskell, what new concepts would they have to learn, what knowledge would they have to disregard, and what syntax would they have to remap?
+
+By teaching concepts we aim o teach fluency.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]:
```suggestion
By teaching concepts we aim to teach fluency.
```

File: docs/concept-exercises.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Concept Exercises
+
+Concept Exercises replace V2's Core Exercises. They are exercises designed to teach specific programming concepts.
+
+## What do we mean by programming concepts?
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]:
```suggestion
## What do we mean by concepts?
```

File: docs/concept-exercises.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Concept Exercises
+
+Concept Exercises replace V2's Core Exercises. They are exercises designed to teach specific programming concepts.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md]:
```suggestion
Concept Exercises replace V2's Core Exercises. They are exercises designed to teach specific (programming) concepts.
```

I think we only need to imply that we are teaching programming concepts once, as that is likely obvious to most.

File: docs/concept-exercises.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+See the C# floating-point-numbers exercise's [hints.md file](./languages/csharp/concept-exercises/numbers-floating-point/.docs/hints.md) for an example.
+
+### `.docs/after.md`
+
+Once the user completes the exercise they will be shown this file, which gives them any bonus information or further reading about the concept taught.
+
+See the C# floating-point-numbers exercise's [after.md file](./languages/csharp/concept-exercises/numbers-floating-point/.docs/after.md) for an example.
+
+### `.docs/debug.md`
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md:68]: I'll do a PR later to remove these here and document them somewhere else.

Comment (iHiD) [reply docs/concept-exercises.md:68]: Great. Thanks :)

**Pull request merged**